lib/molinillo/resolution.rb

```diff
@@ -317,112 +332,57 @@ def attempt_to_activate
         existing_vertex = activated.vertex_named(name)
         if existing_vertex.payload
           debug(depth) { "Found existing spec (#{existing_vertex.payload})" }
-          attempt_to_activate_existing_spec(existing_vertex.payload)
+          attempt_to_filter_existing_spec(existing_vertex)
         else
-          attempt_to_activate_new_spec
+          activate_new_spec
         end
       end
 
-      # Attempts to activate the current {#possibility} (given that it has
-      # already been activated)
+      # Attempts to update the existing vertex's `PossibilitySet` with a filtered version
       # @return [void]
-      def attempt_to_activate_existing_spec(spec)
-        if requirement_satisfied_by?(requirement, activated, spec)
+      def attempt_to_filter_existing_spec(vertex)
+        filtered_set = filtered_possibility_set(vertex)
+        if filtered_set.possibilities.any? &&
+            (vertex.payload.dependencies == possibility.dependencies ||
```

Comment:
since `possibility` is a user-provided object, I think you might need to use `dependencies_for(possibility)` here?